packages/common/src/ProviderUtils.ts

```diff
@@ -119,7 +119,7 @@ function addMnemonicToProvider(
 ): HDWalletProvider {
   const wallet = new HDWalletProvider(mnemonic, provider, keyOffset, numKeys);
   // Sumero-fix: modifying the default polling delay of hdwallet to avoid unnecessary eth calls
-  if (process.env.POLLING_DELAY) {
+  if (Number(process.env.POLLING_DELAY)) {
```

Comment:
how does this help?

Reply:
process.env.POLLING_DELAY returns string value.
Boolean("0") --> true
Boolean(0)    --> false

Reply:
ah, so env was returning string, and was setting a polling delay of 0, and actually never updating the blockchain details. Thanks for explaining.